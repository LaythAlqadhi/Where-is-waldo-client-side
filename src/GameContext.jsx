import React, { useState, createContext } from 'react';
import PropTypes from 'prop-types';

const initialState = {
  state: { loading: false, error: null },
  status: { isStarted: false, isEnded: true },
  message: null,
};

export const GameContext = createContext(initialState);

export function GameProvider({ children }) {
  const [state, setState] = useState({ loading: false, error: null });
  const [status, setStatus] = useState({
    isStarted: false,
    isEnded: false,
    message: null,
  });
  const [message, setMessage] = useState(null);

  return (
    <GameContext.Provider
      value={{
        state,
        setState,
        status,
        setStatus,
        message,
        setMessage,
      }}
    >
      {children}
    </GameContext.Provider>
  );
}

GameProvider.propTypes = {
  children: PropTypes.node.isRequired,
};
