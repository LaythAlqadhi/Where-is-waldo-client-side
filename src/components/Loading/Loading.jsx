import React, { useContext } from 'react';
import { GameContext } from '../../GameContext';

function Loading() {
  const { state } = useContext(GameContext);

  return <div className={`loading ${state.loading && 'active'}`} />;
}

export default Loading;
