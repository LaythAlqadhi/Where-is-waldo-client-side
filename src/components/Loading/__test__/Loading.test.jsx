import React from 'react';
import { render } from '@testing-library/react';
import { GameContext } from '../../../GameContext';
import Loading from '../Loading';

it('renders Loading component correctly', () => {
  const { asFragment } = render(
    <GameContext.Provider value={{ state: { loading: true, error: null } }}>
      <Loading />
    </GameContext.Provider>,
  );

  expect(asFragment()).toMatchSnapshot();
});
