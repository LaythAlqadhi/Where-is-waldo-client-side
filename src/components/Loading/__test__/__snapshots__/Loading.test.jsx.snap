// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders Loading component correctly 1`] = `
<DocumentFragment>
  <div
    class="loading active"
  />
</DocumentFragment>
`;
